File: lib/DevAssist/src/main/java/dev/base/Source.java
package dev.base;

import android.net.Uri;

import java.io.File;

/**
 * detail: 资源路径通用类
 * @author Ttt
 */
public class Source {

    public String mUrl;
    public Uri    mUri;
    public byte[] mBytes;
    public int    mResource;
    public File   mFile;

    private Source() {
    }

    public static Source create(final String url) {
        Source source = new Source();
        source.mUrl = url;
        return source;
    }

    public static Source create(final Uri uri) {
        Source source = new Source();
        source.mUri = uri;
        return source;
    }

    public static Source create(final byte[] bytes) {
        Source source = new Source();
        source.mBytes = bytes;
        return source;
    }

    public static Source create(final int resource) {
        Source source = new Source();
        source.mResource = resource;
        return source;
    }

    public static Source create(final File file) {
        Source source = new Source();
        source.mFile = file;
        return source;
    }

    public static Source createWithPath(final String path) {
        return create(new File(path));
    }
}

File: lib/LocalModules/DevComponent/lib_utils/src/main/java/dev/kotlin/utils/source.kt
package dev.kotlin.utils

import android.graphics.Bitmap
import android.graphics.drawable.Drawable
import android.net.Uri
import dev.base.DevSource
import java.io.File
import java.io.InputStream

// ============
// = DevSource =
// ============

fun String.toSource(): DevSource {
    return DevSource.create(this)
}

fun String.toSourceByPath(): DevSource {
    return DevSource.createWithPath(this)
}

fun Uri.toSource(): DevSource {
    return DevSource.create(this)
}

fun ByteArray.toSource(): DevSource {
    return DevSource.create(this)
}

fun Int.toSource(): DevSource {
    return DevSource.create(this)
}

fun File.toSource(): DevSource {
    return DevSource.create(this)
}

fun InputStream.toSource(): DevSource {
    return DevSource.create(this)
}

fun Drawable.toSource(): DevSource {
    return DevSource.create(this)
}

fun Bitmap.toSource(): DevSource {
    return DevSource.create(this)
}